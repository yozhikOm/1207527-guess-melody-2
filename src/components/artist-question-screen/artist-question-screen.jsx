import React from 'react';
import PropTypes from 'prop-types';

const ArtistQuestionScreen = (props) => {

  const handleFormChange = (evt) => {
    const {question, onAnswer} = props;
    const {answers} = question;

    const i = evt.target.id.split(`-`)[1];
    const userAnswer = answers[i];

    onAnswer(userAnswer);
  };

  const {question, renderPlayer} = props;
  const {answers, song} = question;

  return (
    <section className="game__screen">
      <h2 className="game__title">Кто исполняет эту песню?</h2>
      <div className="game__track">
        <div className="track">
          {renderPlayer(song, 0)}
        </div>
      </div>

      <form className="game__artist" onChange={handleFormChange}>
        {answers.map((it, i) => {
          return (
            <div key={`answer-${i}`} className="artist">
              <input
                className="artist__input visually-hidden"
                type="radio"
                name="answer"
                value={`answer-${i}`}
                id={`answer-${i}`}
              />
              <label className="artist__name" htmlFor={`answer-${i}`}>
                <img className="artist__picture" src={it.picture} alt={it.artist} />
                {it.artist}
              </label>
            </div>
          );
        })}
      </form>
    </section>
  );
};

ArtistQuestionScreen.propTypes = {
  onAnswer: PropTypes.func.isRequired,
  question: PropTypes.shape({
    answers: PropTypes.arrayOf(PropTypes.shape({
      artist: PropTypes.string.isRequired,
      picture: PropTypes.string.isRequired,
    })).isRequired,
    song: PropTypes.shape({
      artist: PropTypes.string.isRequired,
      src: PropTypes.string.isRequired,
    }).isRequired,
    type: PropTypes.oneOf([`genre`, `artist`]).isRequired,
  }).isRequired,
  renderPlayer: PropTypes.func.isRequired,
};

export {ArtistQuestionScreen};
