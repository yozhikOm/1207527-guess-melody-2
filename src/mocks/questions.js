export const questions = [
  {
    type: `genre`,
    genre: `rock`,
    answers: [
      {
        src: `http://www.hochmuth.com/mp3/Vivaldi_Sonata_eminor_.mp3`,
        genre: `classic`,
      },
      {
        src: `https://files.freemusicarchive.org/storage-freemusicarchive-org/music/no_curator/Scott_Holmes/Inspiring__Upbeat_Music/Scott_Holmes_-_05_-_Little_Idea.mp3`,
        genre: `pop`,
      },
      {
        src: `https://files.freemusicarchive.org/storage-freemusicarchive-org/music/no_curator/KieLoKaz/Free_Ganymed/KieLoKaz_-_03_-_Wow_Kielokaz_ID_359.mp3`,
        genre: `jazz`,
      },
      {
        src: `https://files.freemusicarchive.org/storage-freemusicarchive-org/music/ccCommunity/XTaKeRuX/The_way_iam/XTaKeRuX_-_04_-_Lost.mp3`,
        genre: `rock`,
      },
    ],
  },
  {
    type: `genre`,
    genre: `folk`,
    answers: [
      {
        src: `https://files.freemusicarchive.org/storage-freemusicarchive-org/music/Ziklibrenbib/November_Polaroid/st/November_Polaroid_-_05_-_And_Miles_to_Go.mp3`,
        genre: `folk`,
      },
      {
        src: `https://files.freemusicarchive.org/storage-freemusicarchive-org/music/ccCommunity/Chad_Crouch/Motion/Chad_Crouch_-_03_-_Creative_Writing.mp3`,
        genre: `pop`,
      },
      {
        src: `https://files.freemusicarchive.org/storage-freemusicarchive-org/music/no_curator/Checkie_Brown/hey/Checkie_Brown_-_12_-_Rosalie_CB_29.mp3`,
        genre: `jazz`,
      },
      {
        src: `https://files.freemusicarchive.org/storage-freemusicarchive-org/music/no_curator/Lobo_Loco/Salad_Mixed/Lobo_Loco_-_08_-_Sierra_Tracks_ID_1232.mp3`,
        genre: `folk`,
      },
    ],
  },
  {
    type: `artist`,
    song: {
      artist: `Checkie Brown`,
      src: `https://files.freemusicarchive.org/storage-freemusicarchive-org/music/no_curator/Checkie_Brown/hey/Checkie_Brown_-_08_-_Hippie_Bulle_-Stoned_Funghi_CB_28.mp3`,
    },
    answers: [
      {
        picture: `http://placehold.it/134x134`,
        artist: `John Snow`,
      },
      {
        picture: `http://placehold.it/134x134`,
        artist: `Jack Daniels`,
      },
      {
        picture: `http://placehold.it/134x134`,
        artist: `Checkie Brown`,
      },
    ],
  },
];

