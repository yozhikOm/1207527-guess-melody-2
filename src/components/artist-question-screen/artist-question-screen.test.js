import React from 'react';
import {ArtistQuestionScreen} from './artist-question-screen.jsx';
import renderer from 'react-test-renderer';
import {questions} from '../../mocks/questions.js';

it(`ArtistQuestionScreen компонент рендерится корректно`, () => {
  const mockButtonClick = jest.fn();
  const mockQuestion = questions.find((q) => q.type === `artist`);
  const mockRenderPlayer = jest.fn();

  const createNodeMock = () => {
    return {
      oncanplaythrough: null,
      onplay: null,
      onpause: null,
      ontimeupdate: null,
      src: ``
    };
  };

  const artistQuestionScreenComponent = renderer.create(
      <ArtistQuestionScreen
        question={mockQuestion}
        onAnswer={mockButtonClick}
        renderPlayer={mockRenderPlayer}
      />,
      {createNodeMock})
      .toJSON();
  expect(artistQuestionScreenComponent).toMatchSnapshot();
});
