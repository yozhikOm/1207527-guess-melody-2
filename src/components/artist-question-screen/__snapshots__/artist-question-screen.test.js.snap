// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArtistQuestionScreen компонент рендерится корректно 1`] = `
<section
  className="game__screen"
>
  <h2
    className="game__title"
  >
    Кто исполняет эту песню?
  </h2>
  <div
    className="game__track"
  >
    <div
      className="track"
    />
  </div>
  <form
    className="game__artist"
    onChange={[Function]}
  >
    <div
      className="artist"
    >
      <input
        className="artist__input visually-hidden"
        id="answer-0"
        name="answer"
        type="radio"
        value="answer-0"
      />
      <label
        className="artist__name"
        htmlFor="answer-0"
      >
        <img
          alt="John Snow"
          className="artist__picture"
          src="http://placehold.it/134x134"
        />
        John Snow
      </label>
    </div>
    <div
      className="artist"
    >
      <input
        className="artist__input visually-hidden"
        id="answer-1"
        name="answer"
        type="radio"
        value="answer-1"
      />
      <label
        className="artist__name"
        htmlFor="answer-1"
      >
        <img
          alt="Jack Daniels"
          className="artist__picture"
          src="http://placehold.it/134x134"
        />
        Jack Daniels
      </label>
    </div>
    <div
      className="artist"
    >
      <input
        className="artist__input visually-hidden"
        id="answer-2"
        name="answer"
        type="radio"
        value="answer-2"
      />
      <label
        className="artist__name"
        htmlFor="answer-2"
      >
        <img
          alt="Checkie Brown"
          className="artist__picture"
          src="http://placehold.it/134x134"
        />
        Checkie Brown
      </label>
    </div>
  </form>
</section>
`;
